File: spara27/lib/src/models/cliente_model.dart
import 'dart:convert';

Usuario usuarioFromJson(String str) => Usuario.fromJson(json.decode(str));

String usuarioToJson(Usuario data) => json.encode(data.toJson());

class Usuario {
  Usuario(
      {this.apellido,
      this.edad,
      this.email,
      this.nombre,
      this.uid,
      this.urlImage,
      this.idD});

  String? apellido;
  String? edad;
  String? email;
  String? nombre;
  String? uid;
  String? urlImage;
  String? idD;

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        apellido: json["apellido"],
        edad: json["edad"],
        email: json["email"],
        nombre: json["nombre"],
        uid: json["uid"],
        urlImage: json["urlImage"],
      );

  Map<String, dynamic> toJson() => {
        "apellido": apellido,
        "edad": edad,
        "email": email,
        "nombre": nombre,
        "uid": uid,
        "urlImage": urlImage,
      };
}

File: functions/src/models/cliente-model.ts
export interface Cliente {
    idCliente?:      string;
    usuario:         string;
    contraseña:      string;
    nombre:          string;
    apellido:        string;
    edad:            number;
    email:           string;
    fechaNacimiento: string;
    sexo:            string;
}
export function Cliente(data :any, id?:string){
    const { usuario, contraseña, nombre, apellido, edad, email, fechaNacimiento, sexo } = data;

    let object : Cliente = { 
    idCliente: id,
    usuario: usuario,
    contraseña: contraseña,
    nombre: nombre,
    apellido: apellido,
    edad: edad,
    email: email,
    fechaNacimiento:fechaNacimiento,
    sexo: sexo,
    };
    return object;
}